packages/zone/src/heap.js

```diff
@@ -27,16 +30,31 @@ const heapStores = Far('heapStores', {
 });
 
 /**
- * A heap (in-memory) zone that uses the default exo and store implementations.
+ * Create a heap (in-memory) zone that uses the default exo and store implementations.
  *
- * @type {import('.').Zone}
+ * @param {string} [baseLabel]
+ * @returns {import('.').Zone}
  */
-export const heapZone = Far('heapZone', {
-  exoClass: defineExoClass,
-  exoClassKit: defineExoClassKit,
-  exo: makeExo,
-  subZone: (_label, _options) => heapZone,
-  ...heapStores,
-});
+export const makeHeapZone = (baseLabel = 'heapZone') => {
+  const { makeOnce, wrapProvider } = makeOnceKit(baseLabel, detachedHeapStores);
+  return Far('heapZone', {
+    exo: wrapProvider(makeExo),
+    exoClass: wrapProvider(defineExoClass),
+    exoClassKit: wrapProvider(defineExoClassKit),
+    subZone: (label, _options) => {
+      return makeOnce(label, () => makeHeapZone(`${baseLabel}.${label}`));
```

Comment:
Oh, accumulating a dotted path. I did not see that coming, but I like it!

Reply:
Good question! I'm not sure yet how to think about that, but my initial reaction is that we should be concerned, and so remove the prefix. But interested in discussing more.

Reply:
I think prefixing is fine as long as the code has no way to discover / read the prefix. In particular, if the prefix is used in an error, it should be a restricted detail.

Reply:
Aha! I had not noticed that it is diagnostic only. I agree --- a diagnostic-only full path is great.

This is another thing to comment about on the line
```js
  const usedKeys = stores.detached().setStore(`${baseLabel} used keys`);
```
because the non-detached setStore using this string argument in a way visible outside of diagnostics.

Reply:
Done.

Reply:
> Done.

Forgot to push?

Reply:
> Forgot to push?

I don't think so: I added the comment at https://github.com/Agoric/agoric-sdk/pull/7891/files#diff-523ce11d16027d677a0bae03d3886c47cb04085e282b060d4883d0cd7195b633R16-R18

Reply:
Thx. Resolving.